### ISSUE TITLE
[Enhancement]: Add optinal libaries to container to assist in file downloading

### ISSUE THREAD

**Issue by dlsniper** [tubearchivist/tubearchivist#428]
### I've read the documentation

- [X] I'm running the latest version of Tube Archivist and have read the [release notes](https://github.com/tubearchivist/tubearchivist/releases/latest).
- [X] I have read through the [wiki](https://github.com/tubearchivist/tubearchivist/wiki) and the [readme](https://github.com/tubearchivist/tubearchivist#installing-and-updating), particularly the [common errors](https://github.com/tubearchivist/tubearchivist#common-errors) section.

### Operating System

Linux

## Describe the bug
When I look at the logs of the downloader, it mentions that the download speed could be throttled.
To this end, following the instructions should help.

## Steps To Reproduce
Try to download https://www.youtube.com/watch?v=Rh1IQwXy3Y8

## Expected behavior
It should not produce a warning in the logs.

### Relevant log output

```shell
[2023-00-00 00:00:00,000: WARNING/ForkPoolWorker-8] WARNING: [youtube] Rh1IQwXy3Y8: nsig extraction failed: You may experience throttling for some formats
         Install PhantomJS to workaround the issue. Please download it from https://phantomjs.org/download.html
         n = 01HQ3brp2uYEI0 ; player = https://www.youtube.com/s/player/97ea7458/player_ias.vflset/en_US/base.js
```

### Anything else?

Thank you for the project!

**Comment by bbilly1** [comment]
I saw these messages too, but I don't think I'm experiencing any throttling. They usually disappear after an update from the amazing yt-dlp team. I think I only see them from extracting information from from downloading, also I will extract the info correctly and not influence the functionality of TA in any way, just spam the logs...

*Phantom JS* is [no longer maintained](https://phantomjs.org/), last release was January 2016 and is basically shipping a complete 65MB Webkit browser engine as a binary, yes, from 2016. So that's far from ideal on many levels. 

If you are experiencing any throttling, you can always increase the [sleep interval](https://github.com/tubearchivist/tubearchivist/wiki/Settings#downloads), limit your channel rescan schedule to something like once or max twice daily, use a VPN or proxy. But I don't think *Phantom JS* is something we should consider.

**Comment by gitlabt** [comment]
I have this message after downloading 3-5 movies, after that the download speed drops to 40-70 Kbit.
And at the end of this slow download I get a 403 Forbidden error.
And I can't finish downloading any movie because the retry starts from scratch.

I'm using the latest image on the latest debian.
I tried on a regular pc and vm. And the result was the same.

PhantomJS needs a lot of dependencies and I am not able to add it to the image. And tubearchivist only exists as a docker image.

I'm in dead end

Is my only solution is the sleep interval parameter set once every hour for example?

**Comment by bbilly1** [comment]
As mentioned above: VPN/proxy? You can also set the throttle limit to automatically restart the download if your speed falls below a threshold. But if you are getting blocked even outside of TA, then you might need a new public IP.

**Comment by gitlabt** [comment]
How can i set proxy ?
Tubearchivist use users env  https_proxy/http_proxy ?

**Comment by bbilly1** [comment]
I think docker has builtin proxy support: https://docs.docker.com/network/proxy/

Also see here: https://github.com/tubearchivist/tubearchivist/issues/366

**Comment by PhuriousGeorge** [comment]
I get a ton of these messages as well, especially when working on old (ancient) channels.  My personal solution is to export the YTIDs for the videos I have in queue and download them externally on a bare-bones yt-dlp install with phantomjs.  The downloads are still relatively slow, but seem to be a bit faster, then I just import them in.  Personally I'd let TA chug along, but for that to be viable for me, I'd want my auto Rescan task to queue recent videos to the front of my download queue (I know there was a popular request for this, but I'm failing to see it in the features requests - probably just a discord discussion).

I appreciate that we wouldn't necessarily want to add phantomjs to TA, but I do wonder if the ability to side-load mods (phantomjs, aria2, etc) into the docker package (especially if we go single-image) would be a decent proposition.  The Linuxserver.io guys have done this to some extent successfully.  Just an idea, but wanted to include how I handle the perceived problem.

**Comment by bbilly1** [comment]
BTW: fix is incoming: https://github.com/yt-dlp/yt-dlp/issues/6131

Additionally at some point we should implement parallel fragment downloading to speed up these old videos: #316.

**Comment by bbilly1** [comment]
That is now fixed in v0.3.3, please update.

**Comment by dlsniper** [comment]
Thank you for the fix. I can confirm that 0.3.3 solves my issues with the slow downloads.
My apologies for the late reply.
